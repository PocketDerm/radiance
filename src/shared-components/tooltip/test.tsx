import React from 'react';
import { renderer } from 'src/tests/reactTestRendererHelpers';

import { Tooltip } from './index';

describe('<Tooltip />', () => {
  describe('UI snapshot', () => {
    it('renders content and children', () => {
      const component = renderer.create(
        <Tooltip position="bottom" content="Tooltip Content">
          Tooltip Trigger
        </Tooltip>,
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
