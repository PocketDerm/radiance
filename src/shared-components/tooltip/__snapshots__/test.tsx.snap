// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tooltip /> UI snapshot renders content and children 1`] = `
.emotion-9 {
  position: relative;
  -webkit-align-items: left;
  -webkit-box-align: left;
  -ms-flex-align: left;
  align-items: left;
  -webkit-flex-direction: column-reverse;
  -ms-flex-direction: column-reverse;
  flex-direction: column-reverse;
}

.emotion-0 {
  cursor: pointer;
}

.emotion-7 {
  max-width: none;
  top: 120%;
  background: #332e54;
  box-shadow: 0px 8px 24px rgba(51,46,84,0.05);
  border-radius: 0.5rem;
  color: #ffffff;
  min-width: 100px;
  opacity: 0;
  padding: 1rem;
  pointer-events: none;
  position: absolute;
  -webkit-transform: translateY(-8px);
  -ms-transform: translateY(-8px);
  transform: translateY(-8px);
  -webkit-transition-delay: 30ms;
  transition-delay: 30ms;
  -webkit-transition-timing-function: ease-in-out;
  transition-timing-function: ease-in-out;
  -webkit-transition: 350ms;
  transition: 350ms;
  z-index: 5;
  text-align: left;
  font-size: 0.875rem;
  display: block;
}

.emotion-2 {
  z-index: 5;
  position: relative;
}

.emotion-5 {
  position: absolute;
  z-index: 4;
  top: -8px;
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
  left: 50%;
  margin-left: -10px;
}

.emotion-4 {
  display: block;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
  color: #332e54;
  -webkit-transition: color 350ms,-webkit-transform 350ms;
  -webkit-transition: color 350ms,transform 350ms;
  transition: color 350ms,transform 350ms;
}

<div>
  <div
    className="emotion-9 emotion-10"
  >
    <div
      className="emotion-0 emotion-1"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
    >
      Tooltip Trigger
    </div>
    <div
      className="emotion-7 emotion-8"
      open={false}
    >
      <div
        className="emotion-2 emotion-3"
      >
        Tooltip Content
      </div>
      <div
        className="emotion-5 emotion-6"
      >
        <svg
          className="emotion-4"
          data-name="Layer 1"
          fill="#332e54"
          height={16}
          viewBox="0 0 46.29 32.92"
          width={16}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M27.07 30.75a4.63 4.63 0 01-7.85 0L0 0h46.27z"
            fill="#332e54"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
